program-runtime/src/guest_transaction.rs

```diff
@@ -0,0 +1,301 @@
+use {
+    solana_account::ReadableAccount, solana_pubkey::Pubkey, solana_sbpf::ebpf::MM_REGION_SIZE,
+    solana_svm_feature_set::SVMFeatureSet, solana_transaction_context::TransactionContext,
+    std::slice,
+};
+
+/// This is how a slice is represented in the VM.
+/// It should be merged with VmSlice in a future refactor.
+#[repr(C)]
+struct GuestSliceReference {
+    pointer: u64,
+    length: u64,
+}
```

Comment:
This struct can be unified with https://github.com/anza-xyz/agave/blob/7a753dbeeb021f0588b6172eba3cdc15c5672c2c/programs/bpf_loader/src/syscalls/mod.rs#L237-L241.

That requires a somewhat large refactor, so I'll leave it for another PR.